spec/src/main/asciidoc/jakarta-ee.adoc

```diff
@@ -34,8 +88,65 @@ When running in an environment where Jakarta Transactions and Jakarta CDI are av
+
+A Jakarta Data provider that supports Jakarta NoSQL will scan classes marked with the `jakarta.nosql.Entity` annotation.
+
+By supporting Jakarta NoSQL annotations, Jakarta Data providers enable Java developers to utilize familiar and standardized mapping techniques when defining entities in repositories, ensuring compatibility and interoperability with the respective technologies.
+
+=== Jakarta Bean Validation
+
+Integrating with Jakarta Validation ensures data consistency within the Java layer. By applying validation rules to the data, developers can enforce constraints and business rules, preventing invalid or inconsistent information from being processed or persisted.
+
+Incorporating Jakarta Validation into your Jakarta Data applications helps enforce data consistency at the Java layer. Whenever data is converted to or from Java objects, the validation rules defined in the entity are automatically executed. For instance, when persisting a `Student` object, the `name` field will be validated to ensure it is not blank.
+
+Using Jakarta Validation brings several advantages. It helps maintain data integrity, improves data quality, and enhances the reliability of the application. Catching validation errors early in the Java layer can identify and resolve potential issues before further processing or persistence occurs. Additionally, Jakarta Validation allows for declarative validation rules, simplifying the validation logic and promoting cleaner and more maintainable code.
+
+The following code snippet demonstrates the usage of Jakarta Validation annotations in the `Student` entity class:
+
+[source,java]
+----
+@Schema(name = "Student")
+@Entity
+public class Student {
+
+    @Id
+    private String id;
+
+    @Column
+    @NotBlank
+    private String name;
+
+    @Positive
+    @Min(18)
+    @Column
+    private int age;
+}
+----
+
+In this example, the `name` field is annotated with `@NotBlank`, indicating that it must not be blank. The `age` field is annotated with both `@Positive` and `@Min(18)`, ensuring it is a positive integer greater than or equal to 18.
+
+The `School` repository interface, shown below, utilizes the validation rules defined in the `Student` entity:
+
+[source,java]
+----
+@Repository
+public interface School extends PageableRepository<Student, String> {
+
+}
+----
+
+In the Jakarta Data specification, when integrating with Jakarta Bean Validation, the validation process should occur during the insert and update operations, specifically within the save methods of the Repositories interfaces. It means that when performing an insert or update operation on a database, Jakarta Bean Validation will execute the defined constraints to validate the data being inserted or updated.
```

Comment:
A user could do an update operation via `@Query(jpql)` and there is no way that it could be validated because no entity objects are involved, so let's be more precise here and state that validation only applies to inserts/updates that are performed via save/saveAll,
```suggestion
In the Jakarta Data specification, when integrating with Jakarta Bean Validation, the validation process must occur for `save` and `saveAll` methods of Repository interfaces prior to making inserts or updates in the database. It means that when using the save methods to perform an insert or update operation on a database, Jakarta Bean Validation will execute the defined constraints to validate the data being inserted or updated.
```